### PR TITLE
Cached conv ID with twilio sid

### PR DESCRIPTION
**Pull request by chien-wei-lin**

### CODE REVIEW

**Review by ali-ziaei — APPROVED** [review]
Looks good to me. Please wait for  @matthew-ceravolo to approve as he knows requirement for event url for Vonage and twilio.

**Comment by matthew-ceravolo** [comment]
@chien-wei-lin can we add this line?

`status_callback_event=["initiated", "ringing", "answered", "completed"],`

**Review by matthew-ceravolo — APPROVED** [review]

File: vocode/streaming/telephony/conversation/outbound_call.py [hunk 1]
```diff
@@ -144,8 +145,8 @@ async def start(self):
             to_phone=self.to_phone,
             from_phone=self.from_phone,
             record=self.telephony_client.get_telephony_config().record,
-            recording_url=(f"{self.telephony_client.get_telephony_config().recording_url}/{self.conversation_id}"),
-            events_url=self.telephony_client.get_telephony_config().events_url,
+            recording_url=f"{self.telephony_client.get_telephony_config().recording_url}/{self.conversation_id}",
+            events_url=f"{self.telephony_client.get_telephony_config().events_url}/{self.conversation_id}",
```
Comment (matthew-ceravolo) [inline vocode/streaming/telephony/conversation/outbound_call.py:149]: unfortunately, due to the way Twilio handles inbound vs outbound, we wont actually be able to support both this way. we'll still need a generic `/events` endpoint, and we'll need to use the CallSid to lookup the ConversationId

File: vocode/streaming/telephony/client/twilio_client.py [hunk 2]
```diff
@@ -44,8 +52,10 @@ async def create_call(
             record=record,
             recording_status_callback=recording_url,
             status_callback=events_url,
+            status_callback_event=["initiated", "ringing", "answered", "completed"],
             **self.get_telephony_config().extra_params,
         )
+        _redis_client.setex(twilio_call.sid, _ttl_in_seconds, conversation_id)
```
Comment (matthew-ceravolo) [inline vocode/streaming/telephony/client/twilio_client.py:58]: can we prefix this so nothing else tries to store some value in the same key?

**Pull request merged**